examples/amp-web-push/publisher-files/permission-dialog.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+class AmpWebPushPermissionDialog {
+  constructor(options) {
+    if (options && options.debug) {
+      // Debug enables verbose logging for this page and the window and worker
+      // messengers
+      this.debug = true;
+    }
+
+    // For communication between the AMP page and this permission dialog
+    this.ampMessenger = new WindowMessenger({
+      debug: this.debug
+    });
+  }
+
+  isCurrentDialogPopup() {
+    return !!window.opener &&
+      window.opener !== window;
+  }
+
+  requestNotificationPermission() {
+    return new Promise((resolve, reject) => {
+      try {
+        Notification.requestPermission(permission => resolve(permission));
+      } catch (e) {
+        reject(e);
+      }
+    });
+  }
+
+  /**
+   * Tries to decode a URI component, falling back to opt_fallback (or an empty
+   * string)
+   *
+   * @param {string} component
+   * @param {string=} opt_fallback
+   * @return {string}
+   */
+  tryDecodeUriComponent(component, fallback = '') {
```

Comment:
Yep! Now that we have a gulp build task, the function is imported from the AMP framework.